aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2407,12 +2445,15 @@ impl VMValidator for AptosVM {
         let mut session = self.new_session(&resolver, SessionId::prologue_meta(&txn_data));
 
         // Increment the counter for transactions verified.
+        let has_randomness_annotation =
+            check_randomness_annotation(&mut session, &resolver, txn.payload());
```

Comment:
When you do `let txn_data = TransactionMetadata::new(&txn);` few lines above you can probably also add a field there for has_randomness_annotation? Then no need to pass it around. Also calling more generally makes more sense if we want to switch to this flow at some point anyway?

Reply:
we now have `required_deposit` in `TransactionMetadata`.